test/boost/mutation_writer_test.cc

```diff
@@ -60,25 +60,26 @@ SEASTAR_TEST_CASE(test_multishard_writer) {
                 auto muts = gen(partition_nr);
                 std::vector<size_t> shards_before(smp::count, 0);
                 std::vector<size_t> shards_after(smp::count, 0);
+                schema_ptr s = gen.schema();
 
                 for (auto& m : muts) {
-                    auto shard = dht::global_partitioner().shard_of(m.token());
+                    auto shard = s->get_partitioner().shard_of(m.token());
                     shards_before[shard]++;
                 }
-                schema_ptr s = gen.schema();
                 auto source_reader = partition_nr > 0 ? flat_mutation_reader_from_mutations(muts) : make_empty_flat_reader(s);
+                auto& partitioner = s->get_partitioner();
                 size_t partitions_received = distribute_reader_and_consume_on_shards(s,
-                    dht::global_partitioner(),
+                    partitioner,
                     std::move(source_reader),
-                    [&shards_after, error] (flat_mutation_reader reader) mutable {
+                    [&partitioner, &shards_after, error] (flat_mutation_reader reader) mutable {
```

Comment:
How can we pass a schema_ptr by reference? Is it because we assume we run in a seastar::thread and "partitioner" above holds it?

Reply:
We're taking partitioner by refererence - not schema_ptr. BTW this change goes away because schema is enough as you correctly mentioned on one of the previous commits.